### DISCUSSION TITLE
How to change args value from template (Angular)

### DISCUSSION THREAD

**Discussion by jvv** [storybookjs/storybook#18396] [Help]
I'm writing a story with a modal component. I can toggle my isOpen variable from the addon-controls. Trying to figure out how to change the isOpen args from a button click inside the story (use-case for opening a modal). The button itself is not part of the modal component.

I've been playing with useArgs from the api, to no avail

My code so far (not working/rendering at all):
```
const Template = (args) => {
  const [_, updateArgs, resetArgs]  = useArgs();  // removing this and the updateArgs in the method below renders the story
  args.triggerModal = () => {
    updateArgs({isOpen: true});
  }
  
  return {
    props: args,
    template: `
      <div style="width: 240px;">
        <button type="button" class="my-button" [disabled]="isOpen" (click)="triggerModal()">Trigger modal</button>
      </div>
      <my-modal
          [open]="isOpen"
          [zIndex]="100"
          [disableBodyScrollWhenOpen]="true"
          [showCloseButton]="showCloseButton"
          [isBackdropClickable]="isBackdropClickable"
          [headingText]="heading"
          (closed)="onClose($event)"
        >
        <div modal-body [innerHTML]="body | safeHtml"></div>
        <div modal-actions [innerHTML]="actions | safeHtml"></div>
      </my-modal>
  `};
};
```

Im assuming that I miss a very basic/simple piece of the puzzle here. Please help me out :)

**Comment by bozzelliandrea** [comment]
Hi @jvv , i think you are using the React way for create story, check out the angular storybook doc like: https://storybook.js.org/docs/angular/writing-stories/args, i don't know if the code snippet is complete, but you probably miss the module definition (moduleMetadata function) inside the story, also you can use argTypes property to work and play with Angular bindings 

Example: https://github.com/bozzelliandrea/ng-storybook-demo/blob/main/src/stories/task.stories.ts
